리프레쉬 토큰 저장

10kseok opened the pull request:
Auth 서버에서 발급한 리프레쉬 토큰을 레디스에 저장하는 기능을 구현하였습니다.

10kseok committed:
feat: 🎸 추가 SKM 설정 및 암호화된 정보 가져오기

Redis와 Jwt 시크릿을 가져오기 위한 SKM 설정을 마쳤습니다.

10kseok committed:
feat: 🎸 추가 레디스 관련 설정 및 정보 불러오기

리프레쉬 토큰을 저장하기 위해 사용하는 레디스를 pom에 추가해주었습니다.
레디스에 필요한 DB 정보를 담은 빈들을 등록해주었습니다.

10kseok committed:
chore: 🤖 수정 Url 관련 설정 변경

개발시 바로 API 서버에 붙여서 사용했던 URL을 실제 Gateway URL로 옮겼습니다.
ureka 관련 설정도 변경된 사항으로 적용했습니다.

10kseok committed:
chore: 🤖 수정 패키지 이동

jwt 관련 클래스들이 흩어져있어 집중도를 떨어뜨려 jwt 패키지 하나로 관리하게끔 했습니다.

10kseok committed:
feat: 🎸 추가 Jwt 관련 설정 및 정보 불러오기

SKM을 통해 jwt secret를 가져오고 관련 설정을 하는 빈들을 등록해줬습니다.
재발급마다 기존의 것을 지우고 새로 발급하는 것을 고려하여 액세스 토큰은 1시간, 리프레쉬 토큰은 1일로 유효기간을 잡았습니다.

10kseok committed:
feat: 🎸 추가 리프레쉬 토큰 저장소 구현

원본 리프레쉬 토큰을 받아 레디스용 리프레쉬 토큰 객체로 만드는 유틸형 클래스를 작성해주었고,
레디스 레포지토리를 이용해 리프레쉬 토큰의 CRUD를 가능하게하였습니다.

10kseok committed:
feat: 🎸 추가 회원 현재 상태 조회 기능

새로 토큰을 발급하기 전에 현재 상태가 이전 토큰에 담긴 상태와 달라져있을 수 있기 때문에 현재 상태를 조회하는 기능을 구현하였습니다.
Gateway에서 토큰을 검증하므로 Auth 서버도 다른 API 서버와 통신하기 위해 헤더에 토큰을 달아줬습니다.

10kseok committed:
refactor: 💡 수정 리프레쉬 토큰 정보 추가

리프레쉬 토큰을 재발급해줄 때 이전 리프레쉬 토큰에 담긴 정보를 재활용해야하는데 권한이 빠져있었습니다.
권한은 액세스 토큰이 갖고 있었으나 이를 얻기위해선 토큰을 둘 다 받아야해서 헤더 값에서 두 개의 토큰을 받는다면 둘 중 어느 것이 리프레쉬 토큰인지 구별하지 못하는 이슈가 발생될 거라 예상해서 리프레쉬 토큰에 권한을 추가했습니다.

10kseok committed:
feat: 🎸 추가 리프레쉬 토큰 재발급 구현

리프레쉬 토큰을 재발급하는 엔드포인트를 작성해주었고 AuthService를 통하여 재발급합니다.
이전에 발급된 토큰의 재사용을 막기위해 삭제해주었습니다.

10kseok committed:
test: 💍 추가 Jwt 발급 동작 확인 테스트

JwtFactory가 정상적으로 발급되는지에 대한 테스트 코드를 작성하였습니다.

seong1542 commented on src/main/java/store/cookshoong/www/cookshoongauth/config/RedisConfig.java:
```diff
@@ -0,0 +1,50 @@
+
+    @Bean
+    @Profile("!default")
+    public RedisProperties redisProperties(@Value("${cookshoong.skm.keyid.redis}") String redisKeyid,
```
javadoc 달아주세용

seong1542 commented on src/main/java/store/cookshoong/www/cookshoongauth/controller/AuthController.java:
```diff
@@ -46,4 +52,10 @@ public ResponseEntity<LoginSuccessResponseDto> login(@Valid @RequestBody LoginRe
     }
+
+    @GetMapping("/reissue")
+    public ResponseEntity<TokenReissueResponseDto> reissue(@RequestHeader("Authorization") String authorization) {
```
javadoc이 빠진 것 같습니당

seong1542 commented on src/main/java/store/cookshoong/www/cookshoongauth/jwt/JwtConfig.java:
```diff
@@ -1,15 +1,40 @@
+     * @throws JsonProcessingException the json processing exception
+     */
+    @Bean
+//    @Profile("!default")
```
안쓰시는 코드는 삭제해주시면 좋을 것 같습니당

seong1542 reviewed (approved):
수고 많으셨습니다. javadoc 빠진 부분 채워주세요

10kseok committed:
docs: ✏️ 수정 누락된 JavaDoc 작성

누락된 JavaDoc 작성 및 사용하지 않는 코드를 제거했습니다.

10kseok committed:
test: 💍 추가 Controller, Service 테스트 작성

인증처리 및 토큰발급에 대한 Service 코드 내부 코드 동작을 테스트 했습니다.
토큰 재발급에서 헤더값에 대한 테스트와 이전에 테스트 코드에서 바디값이 출력되지 않는 이슈를 해결하였습니다.

10kseok committed:
chore: 🤖 수정 유레카 설정 등록

배포 전 유레카 설정을 다시 등록해주었습니다.
(유레카 설정해제시 HealthController에서 ApplicationManager가 안 잡혀서 Docker Container 생성에서 죽는 이슈가 있었습니다.)
